Update Maid

memothelemo opened the pull request:
This PR is for to update the Maid library to update my `@rbxts/binder` package (which it depends on)
because there is a [code in Binder](https://github.com/Quenty/NevermoreEngine/blob/main/src/binder/src/Shared/Binder.lua#L81) where it uses Maid to cancel any threads at any time.

**There are changes that made to Maid (Nevermore variant):**
- Add thread support for maids (it will cancel a thread upon being destroyed or having an old task to clean up)
- Changed maid error message
- Modified 'task' variable from Maid:DoCleaning to 'job'
- Fix thread cancellation bug from both `Maid.__newindex()` and `Maid:DoCleaning()`

memothelemo committed:
Update Maid from Nevermore engine

There are changes that made to Maid (Nevermore variant):
- Add thread support for maids (it will cancel a thread upon being
  destroyed or having an old task to clean up)
- Changed maid error message
- Modified 'task' variable from Maid:DoCleaning to 'job'

Scyfren commented:
Recent bug fix commit that I think should be added here as well: https://github.com/Quenty/NevermoreEngine/commit/252fc3a5b097dc1a58a998a261b2c28d367839d4

memothelemo commented:
> Recent bug fix commit that I think should be added here as well: https://github.com/Quenty/NevermoreEngine/commit/252fc3a5b097dc1a58a998a261b2c28d367839d4

I'll get into it later. Thanks for letting me know! 😉

memothelemo committed:
Update from Quenty/NevermoreEngine@252fc3a

This is a bug fix from Maid. You can read about this commit given
from GitHub with provided link.
